master/buildbot/process/factory.py

```diff
@@ -92,19 +92,13 @@ def __init__(self, source, configure="./configure",
                  configureEnv=None,
                  configureFlags=None,
                  reconf=None,
-                 compile=None,
-                 test=None,
-                 distcheck=None):
+                 compile=["make", "all"],
```

Comment:
I agree with seankelly.
Its a python felony to initialize method argument with lists.
The correct way is to use a marker object
```python
class Marker(object):
    pass

def my_method(compile=Marker):
     if compile is Marker:
          compile=["make", "all"]
```